File: made_in_python/shell_sort.py
from random import randint
from time import time

the_list = [0] * 12345678

for i in range(len(the_list)): # Python's integer is 64 bit. Becuz of that I made the randomizer like this.
    the_list[i] = randint(-2147483648, 2147483647 - 1) # -1 becuz the endpoint is included

def sort(number_list):
    interval: int = len(number_list) // 2
    temp: int
    j: int

    while interval > 0:
        for i in range(interval, len(number_list)):
            temp = number_list[i]
            j = i

            while j >= interval and number_list[j - interval] > temp:
                number_list[j] = number_list[j - interval]
                j -= interval

            number_list[j] = temp
        interval = interval // 2

start_time: float = time()
sort(the_list)
end_time: float = time()
#print("[" + ", ".join(str(x) for x in the_list) + "]") # To see the array
print((str)(len(the_list)) + " integers sorted in " + (str)((end_time - start_time) * 1000) + " milliseconds.")

File: MadeInJava/src/ShellSort.java
import java.util.Random;
//import java.util.Arrays; // Use this too if you want to see the array

public class ShellSort {
    public static void main(String[] args) {
        int[] theArray = new int[12345678];
        Random r = new Random();

        for (int i = 0; i < theArray.length; i++)
        {
            theArray[i] = r.nextInt();
        }

        long startTime = System.nanoTime();
        sort(theArray);
        long endTime = System.nanoTime();
        //System.out.println(Arrays.toString(theArray)); // To see the array
        System.out.println(theArray.length + " integers sorted in " + ((endTime - startTime) / 1000000.0) + " milliseconds");
    }

    static void sort(int[] numberArray)
    {
        int temp, j;

        for (int interval = numberArray.length / 2; interval > 0; interval /= 2)
        {
            for (int i = interval; i < numberArray.length; i++)
            {
                temp = numberArray[i];

                for (j = i; j >= interval && numberArray[j - interval] > temp; j -= interval)
                {
                    numberArray[j] = numberArray[j - interval];
                }

                numberArray[j] = temp;
            }
        }
    }
}
